### PR TITLE
Enhance testing and build environment

### PR DESCRIPTION
**Pull request by ruflin**
* Use same base environment for manual and automated testing
* Allow to run testsuite with different environment
* Document environments and usage

### CODE REVIEW

**Comment by tsg** [comment]
Can you check if the travis failure is relevant?

**Comment by ruflin** [comment]
@tsg Potentially related as a restart also didn't help. Will look into it. Could be related to Shield in snapshot.

**Comment by ruflin** [comment]
After restarting the build several time it crashes every time but at different stages. Feels like this is a Travis problem? Will investigate further.

**Comment by ruflin** [comment]
This will need a rebase after https://github.com/elastic/beats/pull/1445 is merged

File: .travis.yml [hunk 1]
```diff
@@ -58,6 +58,10 @@ addons:
       - geoip-database
 
 before_install:
+  # Update to most recent docker version
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get update          ; fi
```
Comment (andrewkroh) [inline .travis.yml]: You don't need three separate `if` checks. These can be combined to:

 ```
  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
      sudo apt-get update;
      sudo apt-cache search docker;
      sudo apt-get -o Dpkg::Options::="--force-confnew" install -y docker-engine;
    fi 
```

File: libbeat/scripts/Makefile [hunk 2]
```diff
@@ -31,17 +31,24 @@ GOFILES = $(shell find . -type f -name '*.go')
 GOFILES_NOVENDOR = $(shell find . -type f -name '*.go' -not -path "*/vendor/*")
 SHELL=bash
 ES_HOST?="elasticsearch"
+PWD=$(shell pwd)
```
Comment (andrewkroh) [inline libbeat/scripts/Makefile]: Is this necessary? I think both `PWD` and `CURDIR` should be set by GNU Make.

File: libbeat/scripts/docker-entrypoint.sh [hunk 3]
```diff
@@ -53,7 +53,7 @@ es_url() {
 # green or yellow.
 waitForElasticsearch() {
   echo -n "Waiting on elasticsearch($(es_url)) to start."
-  for ((i=1;i<=60;i++))
+  for ((i=1;i<=120;i++))
```
Comment (andrewkroh) [inline libbeat/scripts/docker-entrypoint.sh]: `i` is being incremented twice in the loop. Remove the second `i++` on line 66 and keep it at 60 seconds.

File: testing/environments/docker/logstash/docker-entrypoint.sh [hunk 4]
```diff
@@ -37,7 +37,7 @@ es_url() {
 # green or yellow.
 waitForElasticsearch() {
   echo -n "Waiting on elasticsearch(${ES_HOST}:${ES_PORT}) to start."
-  for ((i=1;i<=60;i++))
+  for ((i=1;i<=120;i++))
```
Comment (andrewkroh) [inline testing/environments/docker/logstash/docker-entrypoint.sh]: Same comment about the double increment of `i`.

**Comment by andrewkroh** [comment]
LGTM, just a few mostly cosmetic issues.

File: .travis.yml (hunk 1 above)
Comment (ruflin) [inline .travis.yml]: changed

File: libbeat/scripts/Makefile (hunk 2 above)
Comment (ruflin) [inline libbeat/scripts/Makefile]: Will try what happens if I remove it.

**Comment by ruflin** [comment]
Adjustments were pushed.

File: libbeat/scripts/Makefile (hunk 2 above)
Comment (ruflin) [inline libbeat/scripts/Makefile]: I had to readd this line, as without it it breaks. I think the problem is that with the PWD other PWD variable used, during the execution PWD changes. What I need is PWD from the Makefile that is executed.

**Pull request merged**